src/tracing/esp32_trace/esp32_tracing.cpp

```diff
@@ -26,6 +26,113 @@
 namespace chip {
 namespace Tracing {
 namespace Insights {
+namespace {
+static constexpr size_t gPermitListSize = 10;
+struct PermitListEntry
+{
+    size_t hash;
+};
+
+// Implements a murmurhash with 0 seed.
+static size_t MurmurHash(const void * key)
+{
+    const size_t multiplier    = 0x5bd1e995;
+    const size_t shift         = 24;
+    const unsigned char * data = (const unsigned char *) key;
+    size_t hash                = 0;
+
+    while (*data)
+    {
+        size_t value = *data++;
+        value *= multiplier;
+        value ^= value >> shift;
+        value *= multiplier;
+        hash *= multiplier;
+        hash ^= value;
+    }
+
+    hash ^= hash >> 13;
+    hash *= multiplier;
+    hash ^= hash >> 15;
+
+    return hash;
+}
+
+/* PASESession,CASESession,NetworkCommissioning,GeneralCommissioning,OperationalCredentials
+ * are well known permitted entries.
+ */
+static PermitListEntry gPermitList[gPermitListSize] = {
```

Comment:
maybe could be size_t directly.

By the way, is murmur size_t? Not uint64 or uint32? Examples seem to be uint32_t so if that is true we could use that. Also PermitListEntry struct with a single value inside seems useless. Could reduce it to just a typedef if you want to say `using HashValue = uint32_t`